tests/system/source/main.d

```diff
@@ -88,6 +114,143 @@ void main ()
         checkBlockHeight(1);
     }
 
+    /// Make Transactions for block height 2
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        // clients[0] Create frozen transactions and enrolment
+        const Block[] blocks_1 = clients[0].getBlocksFrom(1,1);
+        assert(blocks_1[0].header.height == 1);
+
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            Transaction tx;
+
+            if (keypairs.length > idx)
+            {
+                Transaction freeze_tx = {
+                    type: TxType.Freeze,
+                    inputs: [Input(hashFull(blocks_1[0].txs[idx]), 0)],
+                    outputs: [Output(Amount.MinFreezeAmount,
+                        keypairs[idx].address)]
+                };
+                freeze_tx.inputs[0].signature = kp.secret.sign(hashFull(freeze_tx)[]);
+                clients[0].putTransaction(freeze_tx);
+                freeze_txs ~= freeze_tx;
+            }
+            else
+            {
+                Transaction payment_tx = {TxType.Payment, [], []};
+                payment_tx.inputs ~= Input(hashFull(blocks_1[0].txs[idx]), 0);
+                foreach (idx2; 0 .. Block.TxsInBlock)
+                    payment_tx.outputs ~= Output(Amount.MinFreezeAmount, kp.address);
+
+                payment_tx.inputs[0].signature = kp.secret.sign(hashFull(payment_tx)[]);
+                clients[1].putTransaction(payment_tx);
+                payment_txs ~= payment_tx;
+            }
+        }
+
+        checkBlockHeight(2);
+    }
+
+    /// Make Transactions for block height 3
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        const Block[] blocks_2 = clients[0].getBlocksFrom(0,3);
+        assert(blocks_2[2].header.height == 2);
+
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            Transaction tx = {
+                type: TxType.Payment,
+                inputs: [Input(hashFull(payment_txs[0]), idx)],
+                outputs: [Output(Amount.MinFreezeAmount, kp.address)]
+            };
+
+            auto signature = kp.secret.sign(hashFull(tx)[]);
+            tx.inputs[0].signature = signature;
+            clients[2].putTransaction(tx);
+        }
+
+        checkBlockHeight(3);
+    }
+
+    /// Create enrollments
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        Hash[] utxo_hashs;
+        Pair[] signature_noises;
+        Pair[] node_key_pairs;
+        Enrollment[] enrolls;
```

Comment:
This array is not used after the `foreach` loop, so it's not needed. Just create the enrollment inside of the foreach loop and send it to the client.

Reply:
move to `foreach`